src/MoBi.Core/Commands/SetQuantityPropertyInSimulationCommandBase.cs

```diff
@@ -20,6 +21,11 @@ protected SetQuantityPropertyInSimulationCommandBase(TQuantity quantity, IMoBiSi
          CommandType = AppConstants.Commands.EditCommand;
       }
 
+      protected override void ExecuteWith(IMoBiContext context)
+      {
+         context.Resolve<IQuantityValueInSimulationChangeTracker>().TrackChanges(_quantity, _simulation, () => base.ExecuteWith(context));
```

Comment:
ok that's funky. Calling the base in an action? I have never seen this in my life.

Reply:
The purpose of the action is just so I can wrap it with a step to create the original value before the action if one doesn't exist, then a step to check if the result of the action makes the value equal to the original value.

So, what do you suggest? I only need to do these two steps, and I can definitely unwrap this call and do two separate ones.

Reply:
Ok I understand what this does after reading the code.
Can you create a local variable maybe for the context.Resolve. 
e.g. changeTracker = xxx
changeTracker.TrackChanges()